CVAE_testbed/datasets/synthetic.py

```diff
@@ -37,8 +37,9 @@ def __init__(self, num_batches, BATCH_SIZE, model_kwargs, shuffle=True, corr=Fal
             if self.shuffle is True:
                 while count == 0:
                     C_mask = torch.zeros(C.shape).bernoulli_(0.5)
-                    if len(set([i.item() for i in torch.sum(C_mask, dim = 1)])) == self.model_kwargs['x_dim'] + 1:
-                        count = 1 
+                    count = 1
```

Comment:
So this will only run once?

Reply:
This generates multiple C_masks (each with a 50% probability of zeros) until the mask includes all possible permutations of 0's and 1's